tests/Microsoft.Identity.Test.Unit/PublicApiTests/ExtensiblityTests.cs

```diff
@@ -218,6 +218,121 @@ public async Task ValidateAppTokenProviderAsync()
             }
         }
 
+        [TestMethod]
+        public async Task ValidateAdditionalCacheParametersAreStored()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                              .WithAuthority("https://login.microsoftonline.com/tid/")
+                              .WithClientSecret(TestConstants.ClientSecret)
+                              .WithHttpManager(httpManager)
+                              .BuildConcrete();
+
+                httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+                var result = await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                    .WithAdditionalCacheParameters(new List<string> { "additional_param1", "additional_param2", "additional_param3" })
```

Comment:
Pls add some variability to this test - ensure that the response contains `param1...4` and the test sets only 2 and 3.

Reply:
I have a case like this, I can add another one